### ISSUE TITLE
The `--no-compile` flag and gas report completeness

### ISSUE THREAD

**Issue by davidtperk** [cgewecke/hardhat-gas-reporter#104]
hardhat-gas-reporter produces complete gas reports for `hh test --no-compile` but excludes my contracts for `hh test` run without the `--no-compile` flag. Unable to pinpoint a root cause and don't have a reproducible example yet, but wanted to check if  there are known issues why compilation might cause hardhat-gas-reporter to miss contract calls. Other plugins in use that may be relevant include hardhat-abi-exporter, hardhat-solpp, and hardhat-dependency-compiler.

**Comment by cgewecke** [comment]
> excludes my contracts for hh test run without the --no-compile flag

Oh interesting. Do you mean you only get a partial report and newly compiled contracts are missing? 

The plugin loads compilation artifacts as part of a mocha configuration step (I'd guess this must happen before auto-compilation). And it uses `hre.artifacts` supplied to it by an over-ridden subtask: `TASK_TEST_RUN_MOCHA_TESTS`

(Just leaving a note for myself that this is where we bind that for consumption in the mocha reporter at eth-gas-reporter).
 
https://github.com/cgewecke/hardhat-gas-reporter/blob/104db507a8eb5aaced4cc239093f4bb2d0644afa/src/index.ts#L184

**Comment by davidtperk** [comment]
> Do you mean you only get a partial report and newly compiled contracts are missing?

Precisely. `hh compile && hh test --no-compile` works, but not `hh test`.

`console.log(hre.artifacts._validArtifacts)` just above line 184 returns `[]`with `--no-compile` and a long array of artifacts without `--no-compile`.

**Comment by cgewecke** [comment]
Ah! Thanks for reporting and looking at that. Not sure what to do about this off the top of my head....

**Comment by harunosakura030303-maker** [comment]
This is fixed in hardhat-gas-analytics — the --no-compile flag works properly with gas reporting.

npm install hardhat-gas-analytics

https://www.npmjs.com/package/hardhat-gas-analytics

**Comment by mareek** [comment]
> This is fixed in hardhat-gas-analytics — the --no-compile flag works properly with gas reporting.
> 
> npm install hardhat-gas-analytics
> 
> https://www.npmjs.com/package/hardhat-gas-analytics

This package contains a malware according to [socket.dev analysis](https://socket.dev/npm/package/hardhat-gas-analytics).
Don't download it !
